File: zmq/zguide/hwserver.lua
--
-- Hello World server in lua
-- Binds REP socket to tcp://*:5555
-- Expects "Hello" from client, replies with "World"
--
-- ZeJiang Tang <kasicass@gmail.com>
--

require("zmq")

local context = zmq.init(1)
local socket  = context:socket(zmq.REP)
socket:bind("tcp://*:5555")

while true do
    -- Wait for next request from client
    local request = socket:recv()
    print(string.format("Received request: [%s]", request))
    
    -- Do some 'work'
    -- Well, lua has no native support of sleep(1)

    -- Send reply back to client
    socket:send("World")
end

File: zmq/v4.1.3/zguide/src/samples/hwserver.cpp
#include <zmq.h>
#include <stdio.h>
#include <string.h>
#include <assert.h>
#include <Windows.h>

int main()
{
	void *context = zmq_ctx_new();
	void *responder = zmq_socket(context, ZMQ_REP);
	int rc = zmq_bind(responder, "tcp://*:5555");
	assert(rc == 0);

	while (1)
	{
		char buffer[10];
		zmq_recv(responder, buffer, 10, 0);
		printf("Received Hello\n");
		Sleep(1); // do some 'work'
		zmq_send(responder, "World", 5, 0);
	}

	return 0;
}
